Makefile

```diff
@@ -0,0 +1,3 @@
+generate-mocks:
+	mockgen -source=internal/docker/dockerClient.go -destination=mocks/docker/dockerClient.go
```

Comment:
I don't know what you mean. Where would you call `go:generate`?

Reply:
Instead of having a Makefile you'd add comments such as 
```go
// this generates a mock for the DockerClient interface
//go:generate mockgen -destination=./mocks/docker/dockerClient.go -package=mocks github.com/arcalot/arcaflow-plugin-image-builder/internal/docker DockerClient
```
into your code and run 
```bash
go generate ./...
```  
In essence, this leads to `mockgen` being executed, generating the desired mocks. 
I'd propose to add a `./mocks/generate.go` and bundle all `//go:generate` there (a short entry in the `README.md` for documentation would be good as well).